### ISSUE TITLE
ts VertexAttribute does not match actual type

### ISSUE THREAD

**Issue by mrtnlrsn** [dfpc-coe/node-CoT#36]
Typescript `VertexAttribute` specifies that lat/lon should be `number`.
However they are actually `string` when coming out of `CoT.from_proto()`
```
      detail: {
        shape: {
          polyline: {
            vertex: [
              { _attributes: { lat: '47.7047674', lon: '-122.17861' } },
```

**Comment by ingalls** [comment]
The root cause of this issue appears to be the use of the `Type.Index` feature here(https://github.com/dfpc-coe/node-CoT/blob/main/lib/types.ts#L473).

Under the hood this becomes a `TUnion<[]>` which is represented as an `anyOf` JSONSchema value. Unfortunately AJV generally has oddities around the use of `anyOf`, resulting in the tags not getting parsed.

I'm going to experiment with how best to get this fixed short of just declaring these fields as strings and giving up on type coercion. There are several more places where it would be useful for the coercion to take place.

**Comment by ingalls** [comment]
Draft PR: https://github.com/dfpc-coe/node-CoT/pull/37

**Comment by ingalls** [comment]
This should be fixed in v12.0.0 - Major version due to the data format change and fixes around `archive` tag parsing
